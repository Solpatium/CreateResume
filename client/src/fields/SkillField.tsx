import {Field, IFieldProps, IFieldState} from './Field';
import * as React from 'react';
import { Form, Input, Row, Col, Rate } from 'antd';
const FormItem = Form.Item;

interface ISkillFieldProps extends IFieldProps {
    name?: string,
    rating?: number
}

interface ISkillFieldState extends IFieldState {
    name: string,
    rating: number
}

export default class SkillField extends Field<ISkillFieldProps, ISkillFieldState> {
    constructor(props: ISkillFieldProps) {
        super(props)
        this.state = {
            hidden: false,
            id: `field-${Field.counter}`,
            name: props.name || '',
            rating: props.rating || 1
        }
    }

    public get name() { return 'Skill' };

    public get content(): React.ReactNode {
        return (
            <Row>
                <Col span={12}>
                    <FormItem label="Skill">
                    <Input onChange={this.updateName} placeholder="English" value={this.state.name} style={{ "width" : "90%"}}/>
                    </FormItem>
                </Col>
                <Col span={12}>
                    <FormItem label="Rating">
                    <Rate onChange={this.updateRating} value={this.state.rating}/>
                    </FormItem>
                </Col>
            </Row>)
    }

    public updateRating = (rating: number) => {
        this.setState({rating})
        this.notifyChange()
    }

    public updateName = (event: React.ChangeEvent<HTMLInputElement>) => {
        this.setState({name: event.target.value})
        this.notifyChange()
    }
}