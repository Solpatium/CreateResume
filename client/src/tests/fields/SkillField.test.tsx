import React from 'react';
import SkillField from '../../fields/SkillField';
import renderer from 'react-test-renderer';
import { DragDropContext } from 'react-beautiful-dnd';

test('SkillField renders properly', () => {
  const component = renderer.create(
    <DragDropContext onDragEnd={()=>null}><SkillField index={0}/></DragDropContext>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});