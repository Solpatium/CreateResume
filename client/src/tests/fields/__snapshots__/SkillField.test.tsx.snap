// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillField renders properly 1`] = `
<div
  aria-roledescription="Draggable item. Press space bar to lift"
  className="field"
  data-react-beautiful-dnd-drag-handle="0"
  data-react-beautiful-dnd-draggable="0"
  draggable={false}
  onBlur={[Function]}
  onDragStart={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onTouchStart={[Function]}
  style={
    Object {
      "transform": null,
      "transition": null,
    }
  }
  tabIndex={0}
>
  <div
    className="ant-card ant-card-bordered"
  >
    <div
      className="ant-card-head"
    >
      <div
        className="ant-card-head-wrapper"
      >
        <div
          className="ant-card-head-title"
        >
          Skill
        </div>
        <div
          className="ant-card-extra"
        >
          <div
            className="card-actions"
          >
            <button
              className="ant-btn ant-btn-danger ant-btn-circle ant-btn-icon-only"
              onClick={[Function]}
              type="button"
            >
              <i
                className="anticon anticon-close"
              />
            </button>
            <button
              className="ant-btn ant-btn-circle ant-btn-icon-only"
              onClick={[Function]}
              type="button"
            >
              <i
                className="anticon anticon-down"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ant-card-body"
      style={Object {}}
    >
      <div
        className="ant-row"
        style={undefined}
      >
        <div
          className="ant-col-12"
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-form-item-label"
            >
              <label
                className=""
                htmlFor={null}
                onClick={[Function]}
                title="Skill"
              >
                Skill
              </label>
            </div>
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <input
                    className="ant-input"
                    disabled={false}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="English"
                    style={
                      Object {
                        "width": "90%",
                      }
                    }
                    type="text"
                    value=""
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col-12"
        >
          <div
            className="ant-row ant-form-item"
            style={undefined}
          >
            <div
              className="ant-form-item-label"
            >
              <label
                className=""
                htmlFor={null}
                onClick={[Function]}
                title="Rating"
              >
                Rating
              </label>
            </div>
            <div
              className="ant-form-item-control-wrapper"
            >
              <div
                className="ant-form-item-control"
              >
                <span
                  className="ant-form-item-children"
                >
                  <ul
                    className="ant-rate"
                    onBlur={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onMouseLeave={[Function]}
                    style={Object {}}
                    tabIndex={0}
                  >
                    <li
                      className="ant-rate-star ant-rate-star-full"
                      onClick={[Function]}
                      onMouseMove={[Function]}
                    >
                      <div
                        className="ant-rate-star-first"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                      <div
                        className="ant-rate-star-second"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                    </li>
                    <li
                      className="ant-rate-star ant-rate-star-zero"
                      onClick={[Function]}
                      onMouseMove={[Function]}
                    >
                      <div
                        className="ant-rate-star-first"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                      <div
                        className="ant-rate-star-second"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                    </li>
                    <li
                      className="ant-rate-star ant-rate-star-zero"
                      onClick={[Function]}
                      onMouseMove={[Function]}
                    >
                      <div
                        className="ant-rate-star-first"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                      <div
                        className="ant-rate-star-second"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                    </li>
                    <li
                      className="ant-rate-star ant-rate-star-zero"
                      onClick={[Function]}
                      onMouseMove={[Function]}
                    >
                      <div
                        className="ant-rate-star-first"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                      <div
                        className="ant-rate-star-second"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                    </li>
                    <li
                      className="ant-rate-star ant-rate-star-zero"
                      onClick={[Function]}
                      onMouseMove={[Function]}
                    >
                      <div
                        className="ant-rate-star-first"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                      <div
                        className="ant-rate-star-second"
                      >
                        <i
                          className="anticon anticon-star"
                        />
                      </div>
                    </li>
                  </ul>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
